modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2533,12 +2533,24 @@ def _callable_func(self, func, axis, *args, **kwargs):
     groupby_sum = GroupByReduce.register("sum")
 
     def groupby_mean(self, by, axis, groupby_kwargs, agg_args, agg_kwargs, drop=False):
+        by_columns = by
+        if isinstance(by, BaseQueryCompiler):
+            by_columns = by.columns
+        elif isinstance(by, list):
+            if isinstance(by[0], BaseQueryCompiler):
```

Comment:
why do we only check if the first element of `by`?

Reply:
+1, we should add test coverage for that case

Reply:
Checks all items in by now